frontend/public/components/utils/value-from-pair.jsx

```diff
@@ -51,26 +51,28 @@ const getKeys = (keyMap) => {
   return itemKeys;
 };
 
-const NameKeyDropdownPair = ({name, key, configMaps, secrets, onChange, kind, nameTitle, placeholderString}) => {
+export const NameKeyDropdownPair = ({name, key, configMaps, secrets, onChange, kind, nameTitle, placeholderString, isNameAndKey = true}) => {
   let itemKeys = {};
   let keyRefString;
   const cmItems = {};
   const secretItems = {};
   const nameAutocompleteFilter = (text, item) => fuzzy(text, item.props.name);
   const keyAutocompleteFilter = (text, item) => fuzzy(text, item);
   const keyTitle = _.isEmpty(key) ? <span className="text-muted">Select a key</span> : <span>{key}</span>;
+  const cmResourceString = isNameAndKey ? 'configMapKeyRef' : 'configMapRef';
+  const secretResourceString = isNameAndKey ? 'secretKeyRef' : 'secretRef';
 
   _.each(configMaps.items, (v) => {
-    cmItems[`${v.metadata.name}:configMapKeyRef`] = <ResourceName kind="ConfigMap" name={v.metadata.name} />;
+    cmItems[`${v.metadata.name}:${cmResourceString}`] = <ResourceName kind="ConfigMap" name={v.metadata.name} />;
     if (kind === 'ConfigMap' && _.isEqual(v.metadata.name, name)) {
-      keyRefString = 'configMapKeyRef';
+      keyRefString = cmResourceString;
```

Comment:
nit: I'd call this `refProperty` since it's not always a key ref now (and "property" is more precise than "string")